Source/Core/UICommon/GameFile.cpp

```diff
@@ -353,6 +358,58 @@ bool GameFile::IsElfOrDol() const
   return name_end == ".elf" || name_end == ".dol";
 }
 
+bool GameFile::ReadXMLMetadata(const std::string& path)
+{
+  std::string data;
+  if (!File::ReadFileToString(path, data))
+    return false;
+
+  pugi::xml_document doc;
+  // We are using load_buffer instead of load_file to avoid encoding problems on Windows
+  if (!doc.load_buffer(data.data(), data.size()))
+    return false;
+
+  const pugi::xml_node app_node = doc.child("app");
+  m_pending.custom_name = app_node.child("name").text().as_string();
+  m_pending.custom_maker = app_node.child("coder").text().as_string();
+  m_pending.custom_description = app_node.child("short_description").text().as_string();
+
+  // Elements that we aren't using:
+  // version (can be written in any format)
+  // release_date (YYYYmmddHHMMSS format)
```

Comment:
We do append the revision to the game name (altho this is a `u16` and not just any plain string); and there would be the apploader date field (which _is_ a string)...so we potentially _could_ use those two.

Reply:
I would really not like to change the revision from `u16` to `std::string`. I could do the apploader date, but an apploader date isn't really the same as a release date, and any custom data doesn't show up in the properties dialog anyway the way things are now (which is the only place where you can see the apploader date). Maybe we could rework this if we add a properties dialog for DOL/ELF files at some point.